netmaster/docknet/docknet.go

```diff
@@ -160,23 +164,24 @@ func CreateDockNet(tenantName, networkName, serviceName string, nwCfg *mastercfg
 		ipamOptions["tenant"] = nwCfg.Tenant
 		ipamOptions["network"] = nwCfg.NetworkName
 
+		ipamCfg := network.IPAM{
+			Driver:  ipamDriverName,
+			Config:  ipams,
+			Options: ipamOptions,
+		}
 		// Build network parameters
-		nwCreate := dockerclient.NetworkCreate{
-			Name:           docknetName,
+		nwCreate := types.NetworkCreate{
 			CheckDuplicate: true,
 			Driver:         netDriverName,
-			IPAM: dockerclient.IPAM{
-				Driver:  ipamDriverName,
-				Config:  ipams,
-				Options: ipamOptions,
-			},
-			Options: netPluginOptions,
+			IPAM:           &ipamCfg,
+			Options:        netPluginOptions,
+			Attachable:     true,
 		}
 
 		log.Infof("Creating docker network: %+v", nwCreate)
 
 		// Create network
-		resp, err := docker.CreateNetwork(&nwCreate)
+		resp, err := docker.NetworkCreate(context.Background(), docknetName, nwCreate)
```

Comment:
If this is created in background, would we get error notifications? Or is it guaranteed to eventually succeed?